test/apps/iodemo/io_demo.cc

```diff
@@ -805,16 +817,31 @@ class P2pDemoCommon : public UcxContext {
         return (data_size + chunk_size - 1) / chunk_size;
     }
 
+    static void validate_failure(const UcxConnection *conn,
+                                 const std::stringstream &err_str,
+                                 const char *type_str, size_t length, 
+                                 ucs_memory_type_t mem_type, uint8_t op) {
+        LOG << "ERROR: " << err_str.str() << " detected on "
+            << conn->get_log_prefix() << " (status="
+            << ucs_status_string(conn->ucx_status()) << ") for " << type_str
+            << " (length=" << length << " mem_type=" << mem_type << " op=\""
+            << io_op_names[op] << "\")";
+        abort();
+    }
+
     static void validate(const UcxConnection *conn, const BufferIov& iov,
-                         unsigned seed) {
+                         unsigned seed, io_op_t op) {
+        std::stringstream err_str;
+
         assert(iov.size() != 0);
 
-        size_t err_pos = iov.validate(seed);
+        size_t err_pos = iov.validate(seed, err_str);
         if (err_pos != iov.npos()) {
-            LOG << "ERROR: iov data corruption at " << err_pos
-                << " position detected on " << conn->get_log_prefix()
-                << " (status=" << ucs_status_string(conn->ucx_status()) << ")";
-            abort();
+            std::stringstream err_log_str;
+            err_log_str << "iov data corruption (" << err_str.str() << ") at "
+                        << err_pos << " position";
+            validate_failure(conn, err_log_str, "iov", iov.data_size(),
```

Comment:
do we really need to pass type_str="iov"/"message"? seems it's already printed in the beginning of the error message